File: services/go/admin-bff/internal/models/support.go
package models

import (
	"encoding/json"
	"time"
)

type SupportTicket struct {
	ID                 string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID           int64      `gorm:"not null;index" json:"player_id"`
	PlayerEmail        string     `gorm:"type:varchar(255);not null" json:"player_email"`
	PlayerUsername     string     `gorm:"type:varchar(100);not null" json:"player_username"`
	PlayerGroup        string     `gorm:"type:varchar(50);not null;default:'standard'" json:"player_group"`
	Subject            string     `gorm:"type:varchar(500);not null" json:"subject"`
	Category           string     `gorm:"type:ticket_category;not null;default:'general'" json:"category"`
	Priority           string     `gorm:"type:ticket_priority;not null;default:'normal'" json:"priority"`
	Status             string     `gorm:"type:ticket_status;not null;default:'open'" json:"status"`
	AssignedTo         *string    `gorm:"type:varchar(36)" json:"assigned_to,omitempty"`
	AssignedToName     *string    `gorm:"type:varchar(255)" json:"assigned_to_name,omitempty"`
	CreatedVia         string     `gorm:"type:ticket_created_via;not null;default:'manual'" json:"created_via"`
	SourceChatID       *string    `gorm:"type:varchar(100)" json:"source_chat_id,omitempty"`
	SLAFirstResponseAt *time.Time `json:"sla_first_response_at,omitempty"`
	FirstResponseAt    *time.Time `json:"first_response_at,omitempty"`
	SLAResolveAt       *time.Time `json:"sla_resolve_at,omitempty"`
	ResolvedAt         *time.Time `json:"resolved_at,omitempty"`
	ClosedAt           *time.Time `json:"closed_at,omitempty"`
	LastMessagePreview *string    `gorm:"type:varchar(500)" json:"last_message_preview,omitempty"`
	LastMessageAt      *time.Time `json:"last_message_at,omitempty"`
	MessageCount       int        `gorm:"not null;default:0" json:"message_count"`
	IsSLABreach        bool       `gorm:"not null;default:false" json:"is_sla_breach"`
	CreatedAt          time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt          time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type SupportMessage struct {
	ID           string          `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	TicketID     string          `gorm:"type:uuid;not null;index" json:"ticket_id"`
	AuthorType   string          `gorm:"type:message_author_type;not null" json:"author_type"`
	AuthorID     string          `gorm:"type:varchar(36);not null" json:"author_id"`
	AuthorName   string          `gorm:"type:varchar(255);not null" json:"author_name"`
	IsInternal   bool            `gorm:"not null;default:false" json:"is_internal"`
	Body         string          `gorm:"type:text;not null" json:"body"`
	Attachments  json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"attachments,omitempty"`
	CreatedAt    time.Time       `gorm:"not null;default:now()" json:"created_at"`
}

type TicketLink struct {
	ID             string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	TicketID       string    `gorm:"type:uuid;not null;index" json:"ticket_id"`
	EntityType     string    `gorm:"type:varchar(50);not null" json:"entity_type"`
	EntityID       string    `gorm:"type:varchar(100);not null" json:"entity_id"`
	EntitySummary  *string   `gorm:"type:varchar(255)" json:"entity_summary,omitempty"`
	CreatedAt      time.Time `gorm:"not null;default:now()" json:"created_at"`
}

type SLAConfig struct {
	ID                     string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	Category               string    `gorm:"type:ticket_category;not null;unique" json:"category"`
	FirstResponseMinutes   int       `gorm:"not null;default:30" json:"first_response_minutes"`
	ResolutionMinutes      int       `gorm:"not null;default:240" json:"resolution_minutes"`
	Active                 bool      `gorm:"not null;default:true" json:"active"`
	CreatedAt              time.Time `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt              time.Time `gorm:"not null;default:now()" json:"updated_at"`
}

type AgentWorkload struct {
	ID                    string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	AgentID               string    `gorm:"type:varchar(36);not null;uniqueIndex" json:"agent_id"`
	AgentName             string    `gorm:"type:varchar(255);not null" json:"agent_name"`
	OpenTickets           int       `gorm:"not null;default:0" json:"open_tickets"`
	ResolvedToday         int       `gorm:"not null;default:0" json:"resolved_today"`
	AvgResolutionMinutes  int       `gorm:"not null;default:0" json:"avg_resolution_minutes"`
	UpdatedAt             time.Time `gorm:"not null;default:now()" json:"updated_at"`
}

File: apps/admin/src/types/support.ts
export type TicketCategory = "payment" | "bonus" | "technical" | "account" | "kyc" | "general";
export type TicketPriority = "low" | "normal" | "high" | "urgent";
export type TicketStatus = "open" | "pending_player" | "pending_internal" | "resolved" | "closed";
export type TicketCreatedVia = "chat" | "email" | "manual";

export interface SupportTicket {
  id: string;
  player_id: string;
  player_email: string;
  player_username: string;
  player_group: string;
  subject: string;
  category: TicketCategory;
  priority: TicketPriority;
  status: TicketStatus;
  assigned_to: string | null;
  assigned_to_name: string | null;
  created_via: TicketCreatedVia;
  source_chat_id: string | null;
  sla_first_response_at: string | null;
  first_response_at: string | null;
  sla_resolve_at: string | null;
  resolved_at: string | null;
  closed_at: string | null;
  created_at: string;
  updated_at: string;
  last_message_preview: string | null;
  last_message_at: string | null;
  message_count: number;
  is_sla_breach: boolean;
}

export interface TicketFilters {
  status?: TicketStatus;
  category?: TicketCategory;
  priority?: TicketPriority;
  assigned_to?: string;
  search?: string;
  date_from?: string;
  date_to?: string;
  page?: number;
  page_size?: number;
}

export interface TicketMessage {
  id: string;
  ticket_id: string;
  author_type: "player" | "admin";
  author_id: string;
  author_name: string;
  is_internal: boolean;
  body: string;
  attachments: TicketAttachment[];
  created_at: string;
}

export interface TicketAttachment {
  url: string;
  name: string;
  size: number;
}

export interface TicketLink {
  ticket_id: string;
  entity_type: "withdrawal" | "deposit" | "bonus" | "bet" | "chargeback";
  entity_id: string;
  entity_summary: string;
  created_at: string;
}

export interface CreateTicketPayload {
  player_id: string;
  subject: string;
  category: TicketCategory;
  priority?: TicketPriority;
  body: string;
  created_via?: TicketCreatedVia;
}

export interface SendMessagePayload {
  body: string;
  is_internal?: boolean;
  attachments?: TicketAttachment[];
}

export interface UpdateTicketStatusPayload {
  status: TicketStatus;
  reason?: string;
}

export interface AssignTicketPayload {
  assigned_to: string;
}

export interface LinkEntityPayload {
  entity_type: TicketLink["entity_type"];
  entity_id: string;
}

export interface SlaConfig {
  id: string;
  category: TicketCategory;
  first_response_minutes: number;
  resolution_minutes: number;
  active: boolean;
}

export interface TicketStats {
  total_open: number;
  total_pending_player: number;
  total_pending_internal: number;
  total_resolved_today: number;
  sla_breach_count: number;
  avg_resolution_minutes: number;
  by_category: Record<TicketCategory, number>;
  by_priority: Record<TicketPriority, number>;
}

export interface AgentWorkload {
  agent_id: string;
  agent_name: string;
  open_tickets: number;
  resolved_today: number;
  avg_resolution_minutes: number;
}

export interface SupportTeamDashboard {
  stats: TicketStats;
  agent_workloads: AgentWorkload[];
  sla_breaches: SupportTicket[];
}
